src/NuGet.Core/NuGet.Commands/PackagesLockFileBuilder.cs

```diff
@@ -46,7 +46,7 @@ public PackagesLockFile CreateNuGetLockFile(LockFile assetsFile)
 
                     var dependency = new LockFileDependency()
                     {
-                        Id = library.Name,
+                        Id = library.Name.ToLowerInvariant(),
```

Comment:
This was a tongue in cheek comment. 
I wasn't serious suggesting the package ids should be lowercase, but rather pointing out the absurdity.

Reply:
NuGet specifies that package ids are case insensitive, so I don't think it's absurd. In fact, the v3 folder layout lowercases package ids (for good, different reasons than here).

But doing this will ensure that all dependency ids are consistently lower case, whether they're packages or projects.

Reply:
There's value in casing and that's readability. 
Just because things are case insensitive, does not mean that they shouldn't be case preserving.

We don't automatically lowercase the packageid when the customer specifies it at pack time. 
The assembly names generated by build preserve the casing as well. 

Readability for the vast majority users is more than "weird" source control changes for the minority. There's no "functional failure" here. 

While I understand the argument for projects, as there's a clear and perfectly reasonable way to get into that state, this can absolutely never happen with packages. 
The ID casing is the one from the nuspec. 
And getting a package with a different nuspec file violates the repeatability rule, so it'd fail anyways.